packages/react-router/package.json

```diff
@@ -44,7 +44,7 @@
       }
     },
     "./route-module": {
-      "types": "./dist/production/lib/route-module.d.ts"
+      "types": "./dist/production/lib/types/route-module.d.ts"
```

Comment:
This is the main change. Missed that this was misconfigured earlier due to `any` types causing typegen tests to pass even when they shouldn't. So fixed this, but also fixed the tests to detect this failure mode in the future.